app/genesis.go

```diff
@@ -23,8 +24,27 @@ func NewDefaultGenesisState(cdc codec.Codec) GenesisState {
 	return ModuleBasics.DefaultGenesis(cdc)
 }
 
+// OLD ENCODING CONFIGS (Useful as this is broken currently)
 // EncodingConfig specifies the concrete encoding types to use for a given app.
 // This is provided for compatibility between protobuf and amino implementations.
+// type EncodingConfig struct {
+// 	InterfaceRegistry types.InterfaceRegistry
+// 	Codec             codec.Codec
+// 	TxConfig          client.TxConfig
+// 	Amino             *codec.LegacyAmino
+// }
+
+// // MakeEncodingConfig creates an EncodingConfig for testing
+// func MakeEncodingConfig() EncodingConfig {
+// 	// encodingConfig := MakeEncodingConfig() // infinite loop?? why
```

Comment:
So what it is doing is definining encodingConfig as the result of calling MakeEncodingConfig.  It is a type definition more or less.

tho whyit is then redefined as an interface below, sir that is above my pay grade.